### PR TITLE
[JetBrains] Git-ignore the proto-generated Java files from Supervisor API

### PR DESCRIPTION
**Pull request by felladrin**
## Description

Git-ignore the proto-generated Java files from Supervisor API.

The decision to commit those files was based on the fact that Go and TypeScript generated files were also versioned ([More details in this comment](https://github.com/gitpod-io/gitpod/pull/5687#issuecomment-935893306)). Still, we've noticed there's no actual need for them to be committed, so we can generate them on demand while leaving them git-ignored.

- Note: This PR has two commits, and it should be merged to the main branch like this because the second one is a fix of https://github.com/gitpod-io/gitpod/pull/15273 ([More details in this comment](https://github.com/gitpod-io/gitpod/pull/15468#issuecomment-1362574345))

## How to test

1. Confirm if Werft Build passes.
2. Start a new workspace from this PR, using IntelliJ IDEA, then confirm if smart-completion is working from JetBrains Backend plugin codebase.
3. Open a new workspace from https://github.com/Gitpod-Samples/spring-petclinic, on the Preview Environment of this PR, using IntelliJ IDEA, and confirm if the backend-plugin has been loaded (it's enough to check if our customizations on Control Center are loaded).

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by felladrin** [comment]
I've added a second commit because I noticed the argument was being passed wrongly (should be with "-D" and not with "-P") and also because even running `buildPlugin` gradle task, it was creating the folder.
So the second commit is to fix the Plugin Verifier directory in cache to avoid https://werft.gitpod-dev.com/job/gitpod-build-ide-git-ignore-proto-generated-java-files.7

| With "-P" | With "-D" |
| --- | --- |
| image | image |

**Comment by felladrin** [comment]
Spring PetClinic is working fine in the preview Environment ✅ 

image

When opening Gitpod repo (in the branch of this PR) it correctly starts with Supervisor API Java files in place and smart completion works in JetBrains Plugins ✅ 

image

image

**Review by akosyakov — APPROVED** [review]
/hold

I did not try, rely on testing from @felladrin

File: components/supervisor-api/generate-java.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
Comment (akosyakov) [inline components/supervisor-api/generate-java.sh:1]: It would be nice that we don't need another file. Calling `generate.sh` should do everything.

Comment (felladrin) [reply components/supervisor-api/generate-java.sh:1]: It should, but as it also compiles Go and Typescript files, it takes much longer (15-30 seconds) (and there's also a non-null chance of failing while downloading Go packages).

I could make them only a single file later if the engineering team decides to remove also Go and TypeScript files from versioning.

Comment (akosyakov) [reply components/supervisor-api/generate-java.sh:1]: generate.sh could call generate-java.sh that it works as before

Comment (felladrin) [reply components/supervisor-api/generate-java.sh:1]: Yes, generate.sh is already calling generate-java.sh, so its behaviour remains the same:

image

The reason I left the java compilation in a separate file is for us to be able to call it individually (saving several seconds in the build):

image

image

Comment (akosyakov) [reply components/supervisor-api/generate-java.sh:1]: sounds good

**Comment by felladrin** [comment]
That was a notable reduction of Java files!
| Before | After |
| --- | --- |
| image | image |

**Pull request merged**